lib/wit/gitrepo.py

```diff
@@ -79,8 +80,14 @@ def clone(self, source, name):
             "Trying to clone and checkout into existing git repo!"
         log.info('Cloning {}...'.format(self.name))
 
-        proc = self._git_command("clone", "--no-checkout", source, str(self.path),
-                                 working_dir=str(self.path.parent))
+        cmd = ["clone"]
+        env_ref_path = os.getenv("WIT_WORKSPACE_REFERENCE")
```

Comment:
I  am in general usually a bit reluctant to read environment variables in different "leaf-level" methods within a large program, since it's pretty easy to lose track of them. Even if we do go with environment variables for the user-facing interface, I'd probably prefer centralizing the environment variable in a single module, perhaps adjacent to the command-line argument parsing. I am not familiar, however, enough with Wit to give you any suggestions on how to best thread this data through the Wit code, so I'd probably need @jackkoenig to comment here.

Now, putting that question aside, I think an environment variable as the user interface makes sense here, since it's harmless if it has a bad value (with `-if-able`), and it's kind of nice to set it as a globally exported shell variable so you can use it in multiple places without changing your wit commands. That said, I don't have a strong opinion either way, and we could even do both (CLI option should win, in that case).